### DISCUSSION TITLE
Welcome to transformer-directed-evolution Discussions!

### DISCUSSION THREAD

**Discussion by lucidrains** [lucidrains/transformer-directed-evolution#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by andrinr** [comment]
Hi! I am Andrin and I am currently focused on SciML or more specifically surrogates for physics simulations. I have always been fascinated by alternative (to gradient based) optimization algorithms, such as evolutionary algorithms. Although I have to admit I also have a fair bit of skepticism, IMO at least some degree of mysticism around evolutionary algorithm originates in its naming. 

Is the goal to train a transformer which is responsible for crossover and mutation operations?

Would like to be part of the discussion, if I have time capacity and there is coding to do I am also gladly available to tackle some things. :)

**Reply by lucidrains** [reply]
@andrinr hey Andrin, thanks for your interest

yes, i am interested in optimizing crossover and [selection of the parents](https://github.com/lucidrains/genetic-algorithm-pytorch/blob/main/ga_multi_crossover.py), by attending to the genomes of all the population

**Reply by andrinr** [reply]
Cool. As a small verification project applying the approach to evolve a small neural network on a simple prediction task could already be interesting. Btw. do you expect this approach only works with transformers, or why the fixation with the architecture?

**Reply by lucidrains** [reply]
@andrinr it is simply convenient, as i can attend to all genomes as tokens of flexible population sizes

**Reply by lucidrains** [reply]
besides, hard to bet against transformers given all we've seen so far

**Comment by dagmawibabi** [comment]
For a complete beginner can anyone explain how a Transformer model and RL work together?

**Reply by lucidrains** [reply]
you've heard of [deepseek](https://arxiv.org/abs/2501.12948) by now i'm sure? here's [another](https://arxiv.org/abs/2501.14176) recent paper

**Reply by dagmawibabi** [reply]
How do you go about understanding these research papers. I mean I know you've seen my [Scholarxiv](https://github.com/dagmawibabi/ScholarXIV) project with AI but do you have another approach?

**Reply by lucidrains** [reply]
@dagmawibabi oh wow you are the creator! that's a very nice project 😄

you just have to read a lot of papers until you see through all the jargon and math notation

**Reply by dagmawibabi** [reply]
Thanks a lot, means a lot coming from you. And alright I will keep doing that.
